src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java

```diff
@@ -115,31 +131,57 @@
 
             return Utils.stream(readsIter)
                     .peek(read -> read.setIsDuplicate(false))
+                    .peek(read -> read.setAttribute(MarkDuplicates.DUPLICATE_TYPE_TAG, (String) null))
                     .peek(read -> {
-                // Handle reads that have been marked as non-duplicates (which also get tagged with optical duplicate summary statistics)
-                if (namesOfNonDuplicateReadsAndOpticalCounts.containsKey(read.getName())) {
-                    read.setIsDuplicate(false);
-                    if (markUnmappedMates || !read.isUnmapped()) {
-                        int dupCount = namesOfNonDuplicateReadsAndOpticalCounts.replace(read.getName(), -1);
-                        if (dupCount > -1) {
-                            ((SAMRecordToGATKReadAdapter) read).setTransientAttribute(MarkDuplicatesSparkUtils.OPTICAL_DUPLICATE_TOTAL_ATTRIBUTE_NAME, dupCount);
+                        // Handle reads that have been marked as non-duplicates (which also get tagged with optical duplicate summary statistics)
+                        if (namesOfNonDuplicateReadsAndOpticalCounts.containsKey(read.getName())) {
+                            // If its an optical duplicate, mark it.
+                            if (namesOfNonDuplicateReadsAndOpticalCounts.get(read.getName()) == MARKDUPLICATES_OPPTICAL_DUPLICATE_MARKER) {
```

Comment:
I assume this case only happens when tagging is turned on, otherwise it would never have a read set with optical_duplicate_marker?

Reply:
yes, i figured it wasn't worth threading a defensive check down into here but I can if you woould like

Reply:
No, I think it's fine.  I just wanted to make sure that was the intent... Maybe add a comment explaining that.